news/_posts/2017-11-30-make-back-office-modules-great-again.md

```diff
@@ -0,0 +1,68 @@
+
+### PrestaShop 1.7.3, the module experience re-invented
+
+The first new feature we'd like to present is the Hooks debugger. In all modern pages of Back Office, you have access to the PrestaShop debug toolbar, now enriched by a "Hooks" section.
+
+![PrestaShop Hooks Debugger section](/assets/images/2017/11/hooks_debugger_section.png)
+
+At first look, in every modern page you can know which and how many times hooks have been called. If you access the section panel, you have access to many additional information:
+
+* modules and hooks arguments
+* called and uncalled (but available) hooks
+* which modules react to hooks (the order is respected)
+
+![PrestaShop Hooks Debugger](/assets/images/2017/11/hooks_debugger.png)
+
+This allow every PrestaShop developer to choice wisely the right hook, but also to understand some behaviors if a specific hook doesn't render the expected view without have to enable/disable every module until find the guilty one.
+
+Then, we have introduced two hooks in Product Catalog page to make the contribution of menus simple:
+
+* `displayDashboardToolbarTopMenu` to be able to add new action to the top menu
+* `displayDashboardToolbarIcons` to allow the addition of new icons in Dashboard block (will be available in every Dashboard modern pages)
+
+![New Hooks](/assets/images/2017/11/new_hooks.png)
+
+One more think with PrestaShop 1.7.3, the introduction of **"modules as bundle" system**, allowing the override of templates, services (classes), commands, forms and everything from modern pages in your modules. We are documenting this new system right now, but regarding the templating override we can tell this will work mostly the same way we already work with override in Front Office theme templates.
+
+### PrestaShop 1.7.4 and beyond?
+
+PrestaShop 1.7.4 will use Symfony 3 and will require PHP 5.6 or above. We have decided to stop the support of unmaintained PHP versions for security reason: it's time to upgrade PHP, ideally to the latest version (7.1 at this moment).
+
+There is not broken compatibility in PrestaShop code between PrestaShop 1.7.3 and 1.7.4, and Symfony 3 will allow every developer to work with a PHP framework maintained until 2022.
+
+We will continue to migrate Back Office to Symfony, we want to finish "Advanced Parameters" section and improve our UI Kit since Bootstrap 4 is close to the stable release. We will continue to improve the module developer experience, for instance we are already able to [display more information](https://github.com/PrestaShop/PrestaShop/pull/8557) from modules and hooks or in [our current templates](https://github.com/PrestaShop/PrestaShop/pull/8489) to simplify the overrides.
+
+We are working on a new [form helpers](https://github.com/PrestaShop/PrestaShop/pull/8368) to allow developers to hook into modern forms like they were able to do in PrestaShop 1.6.
```

Comment:
Là, je ne comprends pas trop, excuse-moi. Il s'agit d'offrir la possibilité aux marchants de personnaliser quoi ?

Reply:
Pour les marchands, rien. Pour les développeurs, pouvoir mieux travailler avec les formulaires disponibles sur les pages "modernes/Symfony".

Reply:
"We are also working on new [form helpers](https://github.com/PrestaShop/PrestaShop/pull/8368) so that developers can use better the forms available in the modern pages. Like they were able to do in PrestaShop 1.6."